docs/user-guide/v3_migration.rst

```diff
@@ -0,0 +1,226 @@
+
+- ``N5Store`` - see https://github.com/zarr-developers/n5py for an alternative interface to N5 formatted data.
+- ``ABSStore`` - use the :class:`zarr.storage.FsspecStore` instead along with fsspec's `adlfs backend <https://github.com/fsspec/adlfs>`_.
+- ``DBMStore``
+- ``LMDBStore``
+- ``SQLiteStore``
+- ``MongoDBStore``
+- ``RedisStore``
+
+At present, the latter five stores in this list do not have an equivalent in Zarr-Python 3.
+If you are interested in developing a custom store that targets these backends, see
+:ref:`developing custom stores <user-guide-custom-stores>` or open an
+`issue <https://github.com/zarr-developers/zarr-python/issues>`_ to discuss your use case.
+
+Dependencies
+~~~~~~~~~~~~
+
+When installing using ``pip``:
+
+- The new ``remote`` dependency group can be used to install a supported version of
+  ``fsspec``, required for remote data access.
+- The new ``gpu`` dependency group can be used to install a supported version of
+  ``cuda``, required for GPU functionality.
+- The ``jupyter`` optional dependency group has been removed, since v3 contains no
+  jupyter specific functionality.
+
+Configuration
+~~~~~~~~~~~~~
+
+There is a new configuration system based on `donfig <https://github.com/pytroll/donfig>`_,
+which can be accessed via :mod:`zarr.config`.
+Configuration values can be set using code like the following:
+
+.. code-block:: python
+
+   import zarr
+   zarr.config.set({"array.order": "F"})
+
+Alternatively, configuration values can be set using environment variables,
+e.g. ``ZARR_ARRAY__ORDER=F``.
+
+Configuration options include the following:
+
+- Default Zarr format ``default_zarr_version``
+- Default array order in memory ``array.order``
+- Default codecs ``array.v3_default_codecs`` and ``array.v2_default_compressor``
+- Whether empty chunks are written to storage ``array.write_empty_chunks``
+- Async and threading options, e.g. ``async.concurrency`` and ``threading.max_workers``
+- Selections of implementations of codecs, codec pipelines and buffers
+
+Miscellaneous
+~~~~~~~~~~~~~
+
+- The keyword argument ``zarr_version`` available in most creation functions in :mod:`zarr`
+  (e.g. :func:`zarr.create`, :func:`zarr.open`, :func:`zarr.group`, :func:`zarr.array`) has
+  been deprecated in favor of ``zarr_format``.
+
+🚧 Work in Progress 🚧
+~~~~~~~~~~~~~~~~~~~~~~
+
+Zarr-Python 3 is still under active development, and is not yet fully complete.
+The following list summarizes areas of the codebase that we expect to build out
+after the 3.0 release. If features listed below are important to your use case of
+Zarr-Python, please open (or comment on) a
+`GitHub issue <https://github.com/zarr-developers/zarr-python/issues/new>`_.
+
+- The following functions / methods have not been ported to Zarr-Python 3 yet:
+
+  * :func:`zarr.copy` (:issue:`2407`)
+  * :func:`zarr.copy_all` (:issue:`2407`)
+  * :func:`zarr.copy_store` (:issue:`2407`)
+  * :func:`zarr.Group.move` (:issue:`2108`)
```

Comment:
It's a bit confusing that these say they're not ported, but are linked to functions in the API docs (which would imply they are implemented). Do they need removing from the API docs?

Reply:
All of these exist as stubs in the library but currently raise a `NotImplementedError`

Reply:
Should we remove them from the API docs then, so the docs don't give a misleading impression they exist?

Reply:
No opinion on whether to keep them in the API docs.